### PR TITLE
Taylorr/feature/release specification

### PR DESCRIPTION
**Pull request by Taylorr82**
Staging area for release 0.9.0. After merging, the release will be tagged and released on github releases

### CODE REVIEW

**Comment by Taylorr82** [comment]
This branch has kind of morphed from making the specification to "everything I need to do to release." Which I suppose violates the one thing per pull request rule, but it is all directly related to preparation for release

File: Docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+In each case, when making a version update, decrement the less significant number(s) to 0. As in, 1.4.5 -> 1.5.0.
+
+We do not use -beta suffixes as they are difficult to represent numerically in housekeeping data.
+
+In the software, each version number is kept by a uint16 to fit in the housekeeping data.
+
+### Backwards compatibility
+
+The API must not be made incompatible with previous versions within a major version. If the API becomes backwards incompatible, a new major version should be made. This software does not provide and API in the traditonal sense. API will be taken instead to mean service formats and network protocol formats (for example, MPDU format, sync word). If a feature changes a service such that the format is changed, the major version must be updated.
```
Comment (rcunrau) [inline Docs/releasing.md:24]: "and API" -> "an API"

File: main/version.h [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
 #include <stdint.h>
 
-const char *version = "0.9.0";
-const uint16_t version_major = 0;
-const uint16_t version_minor = 9;
-const uint16_t version_patch = 0;
+#define VERSION "0.9.0"
```
Comment (rcunrau) [inline main/version.h:6]: One advantage of using actual memory is that you can see (and check/extract) the version from the binary without running any code.

File: main/version.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#ifndef VERSION_H
+#define VERSION_H
+
+#include <stdint.h>
+
+const char *version = "0.9.0";
+const uint16_t version_major = 0;
```
Comment (rcunrau) [inline main/version.h]: Any format is fine, although 5 bits per field fits into a single uint16_t.

File: include/FreeRTOSConfig.h [hunk 4]
```diff
@@ -105,7 +105,7 @@
 #define configMAX_TASK_NAME_LEN		  ( 16 )
 #define configIDLE_SHOULD_YIELD		  1
 #define configGENERATE_RUN_TIME_STATS 1
-#define configUSE_MALLOC_FAILED_HOOK  1
+#define configUSE_MALLOC_FAILED_HOOK  0
```
Comment (rcunrau) [inline include/FreeRTOSConfig.h:108]: Not sure what this does, but it reminds me there were times we wanted to add a log message or traceback (from SEGVs) for postmortem analysis.

**Review by rcunrau — APPROVED** [review]

File: include/FreeRTOSConfig.h (hunk 4 above)
Comment (Taylorr82) [reply include/FreeRTOSConfig.h:108]: This is different from that IMO. Malloc's should just be checked for NULL and the satellite continues on. We can always reboot it

File: main/version.h (hunk 3 above)
Comment (Taylorr82) [reply main/version.h]: I'd like to keep it incredibly simple. 3 ints makes sense to me

File: main/version.h (hunk 2 above)
Comment (Taylorr82) [reply main/version.h:6]: One disadvantage is that it's hard to share between files. The compiler didn't like sharing the integers. I suppose a getter would work. What do you think?

**Comment by rcunrau** [comment]
You can share the integers by declaring them in version.h and defining them
in version.c. To avoid multiply defined errors you could put "static" in
front of the declaration, but then they are actually redefined (locally) in
each file that includes version.h

**Pull request merged**